parsl/executors/high_throughput/htex.py

```diff
@@ -0,0 +1,392 @@
+
+    worker_ports : (int, int)
+        Specify the ports to be used by workers to connect to Parsl. If this option is specified,
+        worker_port_range will not be honored.
+
+    worker_port_range : (int, int)
+        Worker ports will be chosen between the two integers provided
+
+    internal_port_range : (int, int)
+        Port range used by Parsl to communicate with internal services.
+
+    """
+
+    def __init__(self,
+                 label='HighThroughputExecutor',
+                 provider=LocalProvider(),
+                 launch_cmd=None,
+                 public_ip="127.0.0.1",
+                 worker_ports=None,
+                 worker_port_range=(54000, 55000),
+                 internal_port_range=(55000, 56000),
+                 storage_access=None,
+                 working_dir=None,
+                 engine_debug=False,
+                 mock=False,
+                 managed=True):
+
+        logger.debug("Initializing HighThroughputExecutor")
+
+        self.label = label
+        self.launch_cmd = launch_cmd
+        self.mock = mock
+        self.provider = provider
+        self.engine_debug = engine_debug
+        self.storage_access = storage_access if storage_access is not None else []
+        if len(self.storage_access) > 1:
+            raise ConfigurationError('Multiple storage access schemes are not yet supported')
+        self.working_dir = working_dir
+        self.managed = managed
+        self.engines = []
+        self.tasks = {}
+
+        self.public_ip = public_ip
+        self.worker_ports = worker_ports
+        self.worker_port_range = worker_port_range
+        self.internal_port_range = internal_port_range
+
+        if not launch_cmd:
+            self.launch_cmd = """process_worker_pool.py {debug} -w {tasks_per_node} --task_url={task_url} --result_url={result_url}"""
+
+    def start(self):
+        """ Here we create the Interchange process and connect to it.
+        """
+        self.outgoing_q = zmq_pipes.TasksOutgoing('tcp://127.0.0.1:50055')
+        self.incoming_q = zmq_pipes.ResultsIncoming('tcp://127.0.0.1:50056')
+
+        self.is_alive = True
+
+        self._queue_management_thread = None
+        self._start_queue_management_thread()
+        self._start_local_queue_process()
+
+        logger.debug("Created management thread : %s", self._queue_management_thread)
+
+        if self.provider:
+            debug_opts = "--debug" if self.engine_debug else ""
+            l_cmd = self.launch_cmd.format(debug=debug_opts,
+                                           task_url=self.worker_task_url,
+                                           result_url=self.worker_result_url,
+                                           tasks_per_node=self.provider.tasks_per_node,
+                                           nodes_per_block=self.provider.nodes_per_block)
+            self.launch_cmd = l_cmd
+            logger.debug("Launch command :{}".format(self.launch_cmd))
```

Comment:
Consider including label of executor in this debug line in case config has multiple executors.